### DISCUSSION TITLE
Fedora install instructions: make them more general to work on rpm-ostree too

### DISCUSSION THREAD

**Discussion by boredsquirrel** [mullvad/mullvadvpn-app#8914] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

Fedora Atomic Desktops are probably the future, they are very beginner friendly and simply work better, as they dont break. But here you have some problems as the dnf commands dont work. Easier would be to just use this command

```
curl https://repository.mullvad.net/rpm/stable/mullvad.repo | sudo tee /etc/yum.repos.d/mullvad.repo
```

and respectively for the unstable repo.

The dnf command for installation also does not apply here, so adding something like that would help:

```
install on Fedora:
   sudo dnf install mullvad-vpn

on rpm-ostree based Fedora:
   rpm-ostree install mullvad-vpn
```

it really is no big deal, but having the dnf-only explanation here makes it one. dnf automatically reads all repos, the config-manager thing is not needed.

Also Fedora and rpm-ostree Fedora are in practice the same, so you dont have to support another Distro or something. Apps are installed, dependencies resolved etc. the same. Just that installing local RPMs is more of a pain and should be avoided at all costs.

If users already have it installed though, here is how to fix it after adding the repo:

```
rpm-ostree uninstall mullvad-vpn --install mullvad-vpn
```

thats it. the local RPM gets removed and the repo one with automatic updates installed.

Before with the local RPMs, people needed to do
```
rpm-ostree uninstall Mullvad-VPN-Version-x.rpm --install Mullvad-VPN-Version-y.rpm
````

which is luckily resolved now.

### Alternative solutions

manual pain, figuring out how to replace the dnf command yourself.

### Type of feature

- [ ] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [X] Other

### Operating System

- [ ] Android
- [ ] iOS
- [ ] Windows
- [ ] macOS
- [X] Linux

**Comment by faern** [comment]
Thank you for these instructions on how to install on rpm-ostree based distributions. We might look into incorporating these in our install guide if we find them suitable for general consumption.

**Comment by boredsquirrel** [comment]
Updated the command, rpm-ostree doesnt need sudo.

Would be great to at least replace the repo-add command, as that works on all Fedora variants and also similarly on OpenSuse.

**Comment by bernado82** [comment]
On openSUSE **Tumbleweed**, MullvadVPN 2025.x can be installed directly _"out-of-the-box"_ from Fedora RPM package (but **NOT on Leap** 15.6, rebuilt still required!).

If you want install MullvadVPN from original Fedora RPM, you can import official **GPG
signing key** with the command:

`sudo rpm --import https://mullvad.net/media/mullvad-code-signing.asc`

or `wget https://mullvad.net/media/mullvad-code-signing.asc`
   (and by importing manually the GPG key file **mullvad-code-signing.asc** with YaST repositories manager)

If you want to get **automatic updates with Zypper** & Discover, you've just to add official Fedora repository with the command (or manually with **YaST**):

`sudo zypper ar -f https://repository.mullvad.net/rpm/stable/x86_64/ MullvadVPN`

and this one if you want install beta versions also:

`sudo zypper ar -f https://repository.mullvad.net/rpm/beta/x86_64/ MullvadVPN_beta`

![Image](https://github.com/user-attachments/assets/2484063b-f620-4541-9326-2e2980b01316)
